### PR TITLE
chore(rxjs-examples.ts): Use new router for nested routes

### PR DESCRIPTION
**Pull request by SekibOmazic**
Use only new router for nested routes and remove the workaround with ng-if

### CODE REVIEW

File: src/app/components/app.ts [hunk 1]
```diff
@@ -94,10 +94,10 @@ let styles   = require('./app.css');
...
 })
 @RouteConfig([
-  { path: '/',              as: 'home',          component: Home },
-  { path: '/dashboard',     as: 'dashboard',     component: Dashboard },
-  { path: '/todo',          as: 'todo',          component: Todo },
-  { path: '/rxjs-examples', as: 'rxjs-examples', component: RxJsExamples }
+  { path: '/',                  as: 'home',          component: Home },
+  { path: '/dashboard',         as: 'dashboard',     component: Dashboard },
+  { path: '/todo',              as: 'todo',          component: Todo },
+  { path: '/rxjs-examples/...', as: 'rxjs-examples', component: RxJsExamples }
```
Comment (gdi2290) [inline src/app/components/app.ts]: oh interesting so this is how `...` works

File: src/app/components/rxjs-examples/rxjs-examples.ts [hunk 2]
```diff
@@ -70,4 +69,15 @@ export class RxJsExamples {
       return 'active';
     }
   }
+
+  /**
+   * Next two methods are not working yet
+   */
+  canActivate() {
+    console.info('canActivate works with the new router')
```
Comment (gdi2290) [inline src/app/components/rxjs-examples/rxjs-examples.ts]: these lifecycle hooks were changed to use annotations [angular2/src/router/lifecycle_annotations_impl.ts](https://github.com/angular/angular/blob/master/modules/angular2/src/router/lifecycle_annotations_impl.ts)

**Comment by gdi2290** [comment]
LGTM, we can refactor the router lifecycle after merge

File: src/app/components/rxjs-examples/rxjs-examples.ts (hunk 2 above)
Comment (SekibOmazic) [inline src/app/components/rxjs-examples/rxjs-examples.ts]: Should read release notes more carefully I guess.

**Comment by SekibOmazic** [comment]
:+1:

**Pull request merged**